lib/Parser/Parse.cpp

```diff
@@ -2252,7 +2301,26 @@ void Parser::AddModuleImportEntry(ModuleImportEntryList* importEntryList, IdentP
     importEntryList->Prepend(*importEntry);
 }
 
-void Parser::AddModuleExportEntry(ModuleExportEntryList* exportEntryList, IdentPtr importName, IdentPtr localName, IdentPtr exportName, IdentPtr moduleRequest)
+void Parser::CheckForDuplicateExportEntry(ModuleExportEntryList* exportEntryList, IdentPtr exportName)
+{
+    Js::PropertyId newPropertyId = EnsurePropertyId(exportName);
```

Comment:
Okay. Two things, then. It's not necessary to verify this via property ID. You can just look at IdentPtr equality, because the parser's hash table will have exactly one entry per identifier. The other is that we should be able to avoid the n^2 search. Is it true that export statements can only occur at global scope?